### PR TITLE
dev-libs/protobuf: add new version 3.19.1

### PR DESCRIPTION
**Pull request by Miezhiko**
- dev-libs/protobuf: add new version 3.19.1
 - dev-python/protobuf-python: add 3.19.1

new version slot is `0/30`

updated patch for `dev-python/protobuf-python`

used portage 3.0.28 for testing

### CODE REVIEW

File: dev-python/protobuf-python/protobuf-python-3.19.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	distutils-r1_python_prepare_all
+}
+
+python_configure_all() {
+	mydistutilsargs=(--cpp_implementation)
+}
+
+python_test() {
+	esetup.py test
+}
```
Comment (arthurzam) [inline dev-python/protobuf-python/protobuf-python-3.19.1.ebuild:59]: We have a more compact way for those, put this one line in global scope before `python_prepare_all`:
`distutils_enable_tests setup.py`

And now you can remove `python_test`

File: dev-python/protobuf-python/protobuf-python-3.19.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+HOMEPAGE="https://developers.google.com/protocol-buffers/ https://github.com/protocolbuffers/protobuf"
+if [[ "${PV}" == "9999" ]]; then
+	SRC_URI=""
+else
+	SRC_URI="https://github.com/protocolbuffers/protobuf/archive/v${PV}.tar.gz -> protobuf-${PV}.tar.gz"
+fi
+
+LICENSE="BSD"
+SLOT="0/30"
+KEYWORDS="~alpha amd64 arm ~arm64 ~hppa ~ia64 ~mips ppc ppc64 ~riscv ~sparc x86 ~amd64-linux ~x86-linux ~x64-macos"
```
Comment (arthurzam) [inline dev-python/protobuf-python/protobuf-python-3.19.1.ebuild:27]: Please don't do straight to stable keywords
All arches should be prefixed with `~`

**Review by arthurzam — CHANGES_REQUESTED** [review]
Thank your contribution.
I'm sorry if the changes list looks long, but let us improve the ebuilds :smile:

Also, please split the commits into 2: one for each package

**Comment by Miezhiko** [comment]
@arthurzam thank you a lot for your feedback, I appreciate your time, attention and fast response.
I will do changes tomorrow.

File: dev-libs/protobuf/protobuf-3.19.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+EAPI=7
+
+inherit autotools elisp-common flag-o-matic multilib-minimal toolchain-funcs
+
+if [[ "${PV}" == "9999" ]]; then
+	inherit git-r3
+
+	EGIT_REPO_URI="https://github.com/protocolbuffers/protobuf"
+	EGIT_SUBMODULES=()
+	SRC_URI=""
```
Comment (arthurzam) [inline dev-libs/protobuf/protobuf-3.19.1.ebuild:13]: We can remove this line `SRC_URI`

File: dev-python/protobuf-python/protobuf-python-3.19.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+inherit distutils-r1
+
+if [[ "${PV}" == "9999" ]]; then
+	inherit git-r3
+
+	EGIT_REPO_URI="https://github.com/protocolbuffers/protobuf"
+	EGIT_CHECKOUT_DIR="${WORKDIR}/protobuf-${PV}"
+	EGIT_SUBMODULES=()
+	SRC_URI=""
```
Comment (arthurzam) [inline dev-python/protobuf-python/protobuf-python-3.19.1.ebuild:15]: We can remove this line `SRC_URI`

File: dev-python/protobuf-python/protobuf-python-3.19.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,52 @@
+# Copyright 2008-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (arthurzam) [inline dev-python/protobuf-python/protobuf-python-3.19.1.ebuild:4]: Empty line after EAPI

```suggestion
EAPI=7

```

**Review by arthurzam — CHANGES_REQUESTED** [review]
Looks much better - excellent work!

Some small last changes.

Also, please sign-off you commits. You can use this command to do it in batch:
`git rebase --exec 'git commit --amend --no-edit -n -s' -i HEAD~2`

**Review by arthurzam — CHANGES_REQUESTED** [review]
Uh, about the name in the sign-off:

By [glep-76](https://www.gentoo.org/glep/glep-0076.html), we must use only our real names in the sign-off. I'm not sure, but I think you aren't using it. In case you prefer not to use the real one, we need your confirmation for it.

For easier read, [this one](https://devmanual.gentoo.org/general-concepts/copyright-policy/index.html) might be useful.

In case you decide to use real name, this command will help to edit the commit message:
`git rebase --exec 'git commit --amend' -i HEAD~3`
In any other case, just a reply with explanation/decision will suffice.

---

Ebuilds are good, and tests passed on my testing container. Excellent job!

**Comment by Miezhiko** [comment]
@arthurzam I don't want to share full name with privacy reason, agree to co-author or fully author-transfer to any developer with granting full authorship but at least mention me in commit message, thought I've got few commits in tree already, would agree to anything to avoid drama.

**Review by arthurzam — APPROVED** [review]
Ok, thank you very much , will merge now :)

**Comment by Miezhiko** [comment]
@arthurzam please help to move this https://github.com/gentoo/gentoo/pull/22875 PR to be in tree too, I've found that existing OpenCV versions will fail to build with this version of protobuf

**Comment by arthurzam** [comment]
@Miezhiko 
I can't merge opencv changes as this isn't my package. But I will cap the version for protobuf in current versions in tree. So the breakage won't hit anybody.

**Comment by Miezhiko** [comment]
Okay, I've rebased my PR.

**Pull request closed without merging**